impeller/typographer/backends/skia/text_render_context_skia.cc

```diff
@@ -184,10 +183,8 @@ static void DrawGlyph(SkCanvas* canvas,
   sk_font.setHinting(SkFontHinting::kSlight);
   sk_font.setEmbolden(metrics.embolden);
 
-  auto glyph_color = has_color ? SK_ColorWHITE : SK_ColorBLACK;
-
   SkPaint glyph_paint;
-  glyph_paint.setColor(glyph_color);
+  glyph_paint.setColor(SK_ColorWHITE);
```

Comment:
This needs to always be SK_ColorWhite, otherwise we end up darkening the dark text on light backgrounds. However this change does fix the issue that non-color glyphs were inconsistently rendered depending on whether or not we had promoted the atlas to full color.

Reply:
IIUC, this is going to make white text look correct, but black text to look too thick, regressing the cases solved for: https://github.com/flutter/flutter/issues/119805 and https://github.com/flutter/flutter/issues/119234.